src/figuresClasses.ts

```diff
@@ -1,11 +1,76 @@
-export interface Figure {}
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a = 0,
+    public b = 0,
+    public c = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides of a triangle must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('must be less than the sum of the other two sides');
```

Comment:
The error message here is incomplete and unclear. Consider rephrasing it to something like: 'Each side of a triangle must be less than the sum of the other two sides.'